.devcontainer/Dockerfile

```
@@ -0,0 +1,15 @@
+ARG VARIANT="3.8"
+FROM mcr.microsoft.com/devcontainers/python:${VARIANT}
+
+#install nox
+RUN python3 -m pip install --upgrade --quiet nox
```

Use a requirements.in and requirements.txt file and [install with hashes](https://pip.pypa.io/en/stable/topics/secure-installs/#secure-installs) for security reasons.

Similar to the following in the release script:

https://github.com/googleapis/python-spanner/blob/7e2e712f9ee1e8643c5c59dbd1d15b13b3c0f3ea/.kokoro/release.sh#L19

https://github.com/googleapis/python-spanner/blob/main/.kokoro/requirements.txt
https://github.com/googleapis/python-spanner/blob/main/.kokoro/requirements.in

```suggestion
COPY requirements.txt /requirements.txt
RUN python3 -m pip install --require-hashes -r  requirements.txt
```